File: ABC/169/b.py
n = int(input())
nums = list(map(int, input().split()))
sum = 1
flag = 1
upper = pow(10, 18)
for num in nums:
  if flag == 1:
    sum = sum * num
    if sum > upper:
      flag = 2
  if num == 0:
    flag = 0
    break
if flag == 2:
  print(-1)
else:
  print(sum * flag)

File: ABC/169/b.c
int main()
{
  int n;
  scanf("%d", &n);
  long a[n];
  long i, sum = 1;
  for (i = 0; i < n; i++)
  {
    scanf("%ld", &a[i]);
    if (a[i] == 0)
    {
      printf("0");
      return 0;
    }
  }
  long max = pow(10, 18);
  for (i = 0; i < n; i++)
  {
    if (sum <= max / a[i])
      sum *= a[i];
    else
    {
      printf("-1");
      return 0;
    }
  }
  printf("%ld", sum);
}
